independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/InjectionPointProvider.java

```diff
@@ -25,9 +25,13 @@ public InjectionPoint get(CreationalContext<InjectionPoint> creationalContext) {
      */
     static InjectionPoint set(InjectionPoint injectionPoint) {
         if (injectionPoint != null) {
-            InjectionPoint prev = InjectionPointProvider.CURRENT.get();
-            InjectionPointProvider.CURRENT.set(injectionPoint);
-            return prev;
+            if (injectionPoint.equals(InjectionPointProvider.CURRENT.get())) {
+                return injectionPoint;
+            } else {
+                InjectionPoint prev = InjectionPointProvider.CURRENT.get();
```

Comment:
You could move that one outside of the `else` and call `InjectionPointProvider.CURRENT.get()` only once in all cases. Or do I miss something?